src/bot.coffee

```diff
@@ -102,149 +121,187 @@ class SlackBot extends Adapter
+      @robot.logger.info "Waiting for reconnect..."
     else
-      @robot.logger.info 'Slack client connection was closed, exiting hubot process'
+      @robot.logger.info "Exiting..."
       @client.disconnect()
+      # NOTE: Node recommends not to call process.exit() but Hubot itself uses this mechanism for shutting down
+      # Can we make sure the brain is flushed to persistence? Do we need to cleanup any state (or timestamp anything)?
       process.exit 1
 
   ###*
   # Slack client received an error
+  #
   # @private
+  # @param {SlackRtmError} error - An error emitted from the [Slack RTM API](https://api.slack.com/rtm)
   ###
   error: (error) =>
-    if error.code is -1
-      return @robot.logger.warning "Received rate limiting error #{JSON.stringify error}"
-    @robot.emit 'error', error
+    @robot.logger.error "Slack RTM error: #{JSON.stringify error}"
+    # Assume that scripts can handle slowing themselves down, all other errors are bubbled up through Hubot
+    # NOTE: should rate limit errors also bubble up?
+    if error.code isnt -1
+      @robot.emit "error", error
 
   ###*
-  # Event received from Slack
+  # Incoming Slack event handler
+  #
+  # This method is used to ingest Slack RTM events and prepare them as Hubot Message objects. The messages are passed
+  # to the Robot#receive() method, which allows executes receive middleware and eventually triggers the various
+  # listeners that are created by scripts.
+  #
+  # Depending on the exact type of event, additional properties may be present. The following describes the "special"
+  # ones that have meaningful handling across all types.
+  #
   # @private
+  # @param {Object} event
+  # @param {string} event.type - this specifies the event type
+  # @param {SlackUserInfo} [event.user] - the description of the user creating this event as returned by `users.info`
+  # @param {SlackConversationInfo} [event.channel] - the description of the conversation where the event happened as
+  # returned by `conversations.info`
+  # @param {SlackBotInfo} [event.bot] - the description of the bot creating this event as returned by `bots.info`
   ###
   eventHandler: (event) =>
     {user, channel} = event
 
     # Ignore anything we sent
-    # NOTE: coupled to getting `rtm.start` data
-    return if user && (user?.id is @self.id)
+    return if user?.id is @self.id
 
     ###*
     # Hubot user object in Brain.
     # User can represent a Slack human user or bot user
-    # 
+    #
     # The returned user from a message or reaction event is guaranteed to contain:
-    # 
+    #
     # id {String}:              Slack user ID
     # slack.is_bot {Boolean}:   Flag indicating whether user is a bot
     # name {String}:            Slack username
     # real_name {String}:       Name of Slack user or bot
     # room {String}:            Slack channel ID for event (will be empty string if no channel in event)
     ###
+    # NOTE: should this be using `updateUserInBrain()`?
```

Comment:
@shanedewael while this PR is about tidying up, not changing the implementation, i wanted to point this out because i felt like it may become a source of bugs. `updateUserInBrain()` does quite a bit of processing to make sure the "shape" of the user stored in the brain fits a specific hubot-compatible shape. if we skip that work here, are we going to break things? can we just call that function here (since it essentially does the same thing)?